File: leetcode/lc-us/cpp/268.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    int missingNumber(vector<int> &nums) {
        int size = (int) nums.size();
        int sum = size * (size + 1) / 2;
        int real_sum = 0;
        for (int num: nums) {
            real_sum += num;
        }
        return sum - real_sum;
    }
};

File: leetcode/lc-us/rust/268.rs
impl Solution {
    pub fn missing_number(nums: Vec<i32>) -> i32 {
        let n = nums.len() + 1;
        let mut sum = n * (n - 1) / 2;
        for &x in nums.iter() {
            sum -= x as usize;
        }
        sum as i32
    }
}
